File: Design Patterns/Behavioral/Strategy/Case 1/Python/DecoyDuck.py
from Duck import Duck
from Fly.FlyBehavior import FlyBehavior
from Quak.QuackBehavior import QuackBehavior
from Fly.NoFly import NoFly
from Quak.Mute import Mute
class DecoyDuck(Duck,FlyBehavior,QuackBehavior):

    def __init__(self) -> None:
        Duck.__init__(self)
        self.setFlyBehavior(NoFly())
        self.setQuakBehavior(Mute())
    
    def fly(self):
        self.getFlyBehavior().fly()

    def quak(self):
        self.getQuakBehavior().quak()

    def dispay():
        pass

    def swim():
        pass

File: Design Patterns/Behavioral/Strategy/Case 1/PHP/DecoyDuck.php
<?php
namespace PHP;
require_once("Quak/QuakBehavior.php");
use PHP\Quak\QuakBehavior;
require_once("Fly/FlyBehavior.php");
use PHP\Fly\FlyBehavior;
require_once("Fly/NoFly.php");
use PHP\Fly\NoFly;
require_once("Quak/Mute.php");
use PHP\Quak\Mute;
require_once("Duck.php");
use PHP\Duck;

class DecoyDuck extends Duck implements QuakBehavior,FlyBehavior{
    
    function __construct()
    {
        $this->setQuakBehavior(new Mute());
        $this->setFlyBehavior(new NoFly());
    }

    function fly()
    {
        $this->getFlyBehavior()->fly();
    }

    function quak()
    {
        $this->getQuakBehavior()->quak();
    }

    function display()
    {
        echo "Displaying DecoyDuck";
    }
}